### PR TITLE
Tooling to generate Checkstyle and Spotbugs suppression files

### PR DESCRIPTION
**Pull request by alzimmermsft**
# Description

Adds `/eng/scripts/linting_suppression_generator.py` which is capable of generating Checkstyle and Spotbugs suppression files for a given project.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/scripts/linting_suppression_generator.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        spotbugs_suppressions.write('                xsi:schemaLocation="https://github.com/spotbugs/filter/3.0.0 https://raw.githubusercontent.com/spotbugs/spotbugs/3.1.0/spotbugs/etc/findbugsfilter.xsd">\n')
+    
+        for type in violations.keys():
+            spotbugs_suppressions.write('  <Match>\n')
+            bugs = sorted(violations[type], key=lambda x: int(x[2]))
+
+            # If there is only one bug, then we can just write it out.
+            if len(bugs) == 1:
+                bug = bugs.pop()
+                spotbugs_suppressions.write(f'    <!-- Line: {bug[2]}, Message: {bug[1]} -->\n')
```
Comment (billwert) [inline eng/scripts/linting_suppression_generator.py:230]: Line number will go stale very quickly, won't it? Might be better to omit it than have it here and confuse future us.

File: eng/scripts/linting_suppression_generator.py [hunk 2]
```diff
@@ -0,0 +1,224 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# Use case: Generates Checkstyle (checkstyle-suppressions.xml) and Spotbugs (spotbugs-exclude.xml) suppression files
+# for the Java SDK.
+#
+# To use this tool, run the following command:
+#
+# python linting_suppression_generator.py --project-folder <project folder>
```
Comment (billwert) [inline eng/scripts/linting_suppression_generator.py:9]: nit: what's a project folder? I assume it is the subdirectories under `sdk\<service>`? One example might help a lot here.

File: eng/scripts/linting_suppression_generator.py (hunk 1 above)
Comment (alzimmermsft) [reply eng/scripts/linting_suppression_generator.py]: This has been removed

File: eng/scripts/linting_suppression_generator.py (hunk 2 above)
Comment (alzimmermsft) [reply eng/scripts/linting_suppression_generator.py:9]: Will add that, it'd be `sdk\core\azure-core` not `sdk\core`

**Review by billwert — APPROVED** [review]

**Pull request merged**